Unhandled Error

After a long series of

```
Succesful connection established
Disconnected from server.Trying to reconnect in 1 seconds. 1
Succesful connection established
...
```

The client died:

```
Succesful connection established
/home/ubuntu/tunnel-client/node_modules/ws/lib/WebSocket.js:218
    else throw new Error('not opened');
         ^

Error: not opened
    at WebSocket.send (/home/ubuntu/tunnel-client/node_modules/ws/lib/WebSocket.js:218:16)
    at WebSocket.sendJson (/home/ubuntu/tunnel-client/node_modules/websocket-monkeypatch/lib/index.js:12:15)
    at WebSocket.sendCommand (/home/ubuntu/tunnel-client/node_modules/websocket-monkeypatch/lib/index.js:24:15)
    at WebSocket.<anonymous> (/home/ubuntu/tunnel-client/src/index.js:29:8)
    at emitNone (events.js:67:13)
    at WebSocket.emit (events.js:166:7)
    at WebSocket.establishConnection (/home/ubuntu/tunnel-client/node_modules/ws/lib/WebSocket.js:862:8)
    at ClientRequest.upgrade (/home/ubuntu/tunnel-client/node_modules/ws/lib/WebSocket.js:753:25)
    at ClientRequest.g (events.js:260:16)
    at emitThree (events.js:97:13)

npm ERR! Linux 3.13.0-48-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "start"
npm ERR! node v4.2.4
npm ERR! npm  v2.14.12
npm ERR! code ELIFECYCLE
npm ERR! tunnel-server@0.2.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the tunnel-server@0.2.0 start script 'node index.js'.
npm ERR! This is most likely a problem with the tunnel-server package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node index.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls tunnel-server
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ubuntu/tunnel-client/npm-debug.log
```

Would be useful to see what was happening on the server. I think that the reason we it was being kicked out was because of [this block](https://github.com/4Catalyzer/tunnel-server/blob/master/server/lib/network-registry.js#L28-L32). This reaises the question of why the network was still registered, becuase I would expect the previous network to be deregistered by [this monitor](https://github.com/4Catalyzer/tunnel-server/blob/master/server/lib/network-registry.js#L37) Still, it doesn't explain why the client failed after some attempts.

--- a/src/index.js
+++ b/src/index.js
@@ -24,9 +24,11 @@ function connect() {
   ws = new WebSocket(serverUrl, { headers });
 
   ws.on('open', () => {
-    log('Succesful connection established');
-    retries = 0;
-    ws.sendCommand('REGISTER_NETWORK', networkId);
+    if (this._wsConn.readyState === this._wsConn.OPEN) {
+      log('Succesful connection established');
+      retries = 0;
+      ws.sendCommand('REGISTER_NETWORK', networkId);
+    }
   });
 
   ws.on('close', (code, message) => log('Connection closed.', code, message));

